flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -152,6 +153,9 @@ def is_node(x: tp.Any) -> bool:
 def is_node_type(x: type[tp.Any]) -> bool:
   return x in NODE_TYPES or x is PytreeType
 
+def is_state_leaf(x: tp.Any) -> bool:
+  return isinstance(x, (Variable, np.ndarray, jax.Array))
```

Comment:
Am I right that you're intentionally disallowing Python scalars here?

Reply:
Main problem here is that `Variable` is registered as a pytree so it would get accepted but we explicitly want to forbid it.